crates/common/checkpoint_sync/src/lib.rs

```diff
@@ -0,0 +1,206 @@
+        type Value = bool;
+
+        fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+            formatter.write_str("a boolean or a string representing a boolean")
+        }
+
+        fn visit_bool<E>(self, v: bool) -> Result<Self::Value, E> {
+            Ok(v)
+        }
+
+        fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+        where
+            E: serde::de::Error,
+        {
+            v.parse::<bool>().map_err(E::custom)
+        }
+    }
+
+    deserializer.deserialize_any(BoolVisitor)
+}
+
+fn option_bool_from_str_or_bool<'de, D>(deserializer: D) -> Result<Option<bool>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    Ok(Some(bool_from_str_or_bool(deserializer)?))
+}
+
+/// Entry point for checkpoint sync.
+pub async fn initialize_db_from_checkpoint(
+    db: ReamDB,
+    checkpoint_sync_url: Option<Url>,
+) -> anyhow::Result<()> {
+    if db.is_initialized() {
+        warn!("Starting Checkpoint Sync from existing DB. It is advised to start from a clean DB");
+        return Ok(());
+    }
+
+    let checkpoint_sync_url = get_checkpoint_sync_sources(checkpoint_sync_url).remove(0);
+    info!("Initiating Checkpoint Sync");
+    let block = fetch_finalized_block(&checkpoint_sync_url).await?;
+    info!(
+        "Downloaded Block: {} with Root: {}. Slot: {}",
+        block.data.message.body.execution_payload.block_number,
+        block.data.message.block_root(),
+        block.data.message.slot
+    );
+    let slot = block.data.message.slot;
+    initialize_blobs_in_db(
+        &checkpoint_sync_url,
+        db.clone(),
+        block.data.message.block_root(),
+    )
+    .await?;
+    info!(
+        "Downloaded blobs for block: {}",
+        block.data.message.body.execution_payload.block_number
+    );
+
+    let state = get_state(&checkpoint_sync_url, slot).await?;
+    info!(
+        "Downloaded State with root: {}. Slot: {}",
+        state.data.state_root(),
+        slot
+    );
+    ensure!(block.data.message.slot == state.data.slot, "Slot mismatch");
+
+    ensure!(block.data.message.state_root == state.data.state_root());
+    let mut store = get_forkchoice_store(state.data, block.data.message, db)?;
+
+    let time = network_spec().genesis.genesis_time + SECONDS_PER_SLOT * (slot + 1);
+    on_tick(&mut store, time)?;
+    info!("Initial Sync complete");
+
+    Ok(())
+}
+
+/// Fetch initial state from trusted RPC
+pub async fn get_state(
+    rpc: &Url,
+    slot: u64,
+) -> anyhow::Result<BeaconVersionedResponse<BeaconState>> {
+    let client = reqwest::Client::new();
+    let state = client
+        .get(format!("{rpc}eth/v2/debug/beacon/states/{slot}"))
+        .header(ACCEPT, HeaderValue::from_static("application/octet-stream"))
+        .send()
+        .await?
+        .bytes()
+        .await?;
+
+    Ok(BeaconVersionedResponse::new(
+        BeaconState::from_ssz_bytes(&state).expect("Unable to decode from SSZ"),
+    ))
+}
```

Comment:
```suggestion
/// Fetch initial state from trusted RPC
pub async fn get_state(
    rpc: &Url,
    slot: u64,
) -> anyhow::Result<BeaconState> {
    let client = reqwest::Client::new();
    let state = client
        .get(format!("{rpc}eth/v2/debug/beacon/states/{slot}"))
        .header(ACCEPT, HeaderValue::from_static("application/octet-stream"))
        .send()
        .await?
        .bytes()
        .await?;

    Ok(BeaconState::from_ssz_bytes(&state).expect("Unable to decode from SSZ"),
    )
}
```

can you remove the expect and use a `?` as well